### PR TITLE
[receiver/filelog] added example and additional information for configs

### PR DESCRIPTION
**Pull request by ZahidMirza95**
**Description:** <Describe what has changed.>
Added example on how to use filelog receiver with [ADOT](https://github.com/aws-observability/aws-otel-collector). Also, updated some of the configurations to clarify unit that they're in (ex: specifying they're in bytes) and also specified limitations for some of the configurations. Also, specified default value for multiline to clarify that by default logs will be split by newline character.

**Link to tracking Issue:** 

**Testing:** N/A

**Documentation:** 

- Example on how to use file log receiver with ADOT
- Specifying units and limitations for some configurations
- Changed the default value of multiline configuration to "\n", since logs will be split by newline characters by default

### CODE REVIEW

File: receiver/filelogreceiver/README.md [hunk 1]
```diff
@@ -111,4 +111,66 @@ receivers:
           parse_from: attributes.time
           layout: '%Y-%m-%d %H:%M:%S'
 ```
+## Example - Using filelog receiver with [ADOT](https://github.com/aws-observability/aws-otel-collector) (AWS Distro for OpenTelemetry)
+DISCLAIMER: ADOT doesn't officially support the filelog receiver yet, so the below example is a temporary workaround to get it working
```
Comment (rapphil) [inline receiver/filelogreceiver/README.md:115]: This is the contrib repository, so no mentions to ADOT are necessary.

Comment (rapphil) [reply receiver/filelogreceiver/README.md:115]: Please remove all mentions to ADOT. All the configurations refer to this component, therefore all references should be towards this component alone. 

E.g.:  This example will show how to get the filelog receiver working.

File: receiver/filelogreceiver/README.md [hunk 2]
```diff
@@ -17,37 +17,37 @@ Tails and parses logs from files.
...
+| `exclude`                           | []                                   | A list of file glob patterns to exclude from reading                                                                                                                                                                                                                                        |
+| `start_at`                          | `end`                                | At startup, where to start reading logs from the file. Options are `beginning` or `end`                                                                                                                                                                                                     |
+| `multiline`                         | `\n`                                  | A `multiline` configuration block. See below for more details.                                                                                                                                                                                                                              |
```
Comment (djaglowski) [inline receiver/filelogreceiver/README.md:25]: This is incorrect.

File: receiver/filelogreceiver/README.md [hunk 3]
```diff
@@ -17,37 +17,37 @@ Tails and parses logs from files.
...
+| `include_file_name_resolved`        | `false`                              | Whether to add the file name after symlinks resolution as the attribute `log.file.name_resolved`.                                                                                                                                                                                           |
+| `include_file_path_resolved`        | `false`                              | Whether to add the file path after symlinks resolution as the attribute `log.file.path_resolved`.                                                                                                                                                                                           |
+| `poll_interval`                     | 200ms                                | The duration between filesystem polls in milliseconds.                                                                                                                                                                                                                                      |
```
Comment (djaglowski) [inline receiver/filelogreceiver/README.md:34]: The user must specify the unit of time, which is not necessarily milliseconds.

File: receiver/filelogreceiver/README.md [hunk 4]
```diff
@@ -17,37 +17,37 @@ Tails and parses logs from files.
...
+| `fingerprint_size`                  | `100 bytes (1kb)`                    | The number of bytes with which to identify a file. The first bytes in the file are used as the fingerprint. Decreasing this value at any point will cause existing fingerprints to forgotten, meaning that all files will be read from the beginning (one time). Minimum value is 16 bytes. |
```
Comment (djaglowski) [inline receiver/filelogreceiver/README.md:35]: ?

File: receiver/filelogreceiver/README.md [hunk 5]
```diff
@@ -17,37 +17,37 @@ Tails and parses logs from files.
...
+| `max_log_size`                      | `1048576 bytes (1MiB)`               | The maximum size of a log entry to read in bytes. A log entry will be truncated if it is larger than `max_log_size`. Protects against reading large amounts of data into memory.                                                                                                            |
```
Comment (djaglowski) [inline receiver/filelogreceiver/README.md:36]:
```suggestion
| `max_log_size`                      | `1048576 bytes (1MiB)`               | The maximum size of a log entry to read in bytes. A log entry will be truncated if it is larger than `max_log_size`. Protects against reading large amounts of data into memory.                                                                                                            |
```

This is unnecessary. It's not accurate to state that the unit is bytes.

File: receiver/filelogreceiver/README.md [hunk 6]
```diff
@@ -17,37 +17,37 @@ Tails and parses logs from files.
...
+| `max_concurrent_files`              | 1024                                 | The maximum number of log files from which logs will be read concurrently. If the number of files matched in the `include` pattern exceeds this number, then files will be processed in batches. Minimum value is 1.                                                                        |
+| `max_batches`                       | 0                                    | Only applicable when files must be batched in order to respect `max_concurrent_files`. This value limits the number of batches that will be processed during a single poll interval. A value of 0 indicates no limit.                                                                       |
+| `delete_after_read`                 | `false`                              | If `true`, each log file will be read and then immediately deleted. Requires that the `filelog.allowFileDeletion` feature gate is enabled. Cannot use if start_at is set to "end".                                                                                                          |
```
Comment (djaglowski) [inline receiver/filelogreceiver/README.md:39]:
```suggestion
| `delete_after_read`                 | `false`                              | If `true`, each log file will be read and then immediately deleted. Requires that the `filelog.allowFileDeletion` feature gate is enabled. Cannot use if `start_at` is set to "end".                                                                                                          |
```

File: receiver/filelogreceiver/README.md [hunk 7]
```diff
@@ -17,37 +17,37 @@ Tails and parses logs from files.
...
+| `operators`                         | []                                   | An array of [operators](../../pkg/stanza/docs/operators/README.md#what-operators-are-available). See below for more details                                                                                                                                                                 |
+| `storage`                           | none                                 | The ID of a storage extension to be used to store file checkpoints. File checkpoints allow the receiver to pick up where it left off in the case of a collector restart. If no storage extension is used, the receiver will manage checkpoints in memory only.                              |
+| `header`                            | nil                                  | Specifies options for parsing header metadata. Requires that the `filelog.allowHeaderMetadataParsing` feature gate is enabled. Cannot use if start_at is set to "end". See below for details.                                                                                               |
```
Comment (djaglowski) [inline receiver/filelogreceiver/README.md:44]:
```suggestion
| `header`                            | nil                                  | Specifies options for parsing header metadata. Requires that the `filelog.allowHeaderMetadataParsing` feature gate is enabled. Cannot use if `start_at` is set to "end". See below for details.                                                                                               |
```

File: receiver/filelogreceiver/README.md (hunk 1 above)
Comment (djaglowski) [reply receiver/filelogreceiver/README.md:115]: Seconded. This documentation concerns this receiver, not ADOT

**Review by djaglowski — CHANGES_REQUESTED** [review]

**Comment by djaglowski** [comment]
Please sign the CLA

**Pull request closed without merging**